### PR TITLE
Update Using Git In The Real World to reflect other changes in the curriculum Git content

### PR DESCRIPTION
**Pull request by wise-king-sullyman**
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

TheOdinProject/top-meta/issues/63

**2. This PR:**

Updates the curriculum content as specified in that issue.

**3. Additional Information:**

### CODE REVIEW

File: ruby_programming/git/lesson_using_git_in_the_real_world.md [hunk 1]
```diff
@@ -1,101 +1,79 @@
...
 
-The thing about Git is that, unless you've got a seriously impressive memory, you can't just learn it by reading about it up front... you need to do it.  Find a problem you want to go back and fix, hit an error in your merge, etc. and Google the hell out of it, learning a new Git tactic in the process.  
-
-To help you out, come back and refer to this lesson again when you're in trouble. We'll first cover a real-world example of a GitHub workflow used on this very project.  The Additional Resources section below should also help you find high quality resources for when you need them later on.
+* Using Git to make open source contributions
 
 ### A Git Workflow For Open Source Contribution
 
-Let's say you want to contribute to the web application that powers this website (check it out [here](https://github.com/TheOdinProject/theodinproject)).
+Let's say you want to contribute to the web application that powers this website (it's a different repo than our curriculum content, you can check it out [here](https://github.com/TheOdinProject/theodinproject)).
```
Comment (01zulfi) [inline ruby_programming/git/lesson_using_git_in_the_real_world.md:17]: We could improve the link text here

File: ruby_programming/git/lesson_using_git_in_the_real_world.md [hunk 2]
```diff
@@ -1,101 +1,79 @@
...
 
+1. Create a new feature branch for whatever feature you want to build, and add commits following the same practices that you learned about in the [branching section of our Revisiting Rock Paper Scissors lesson](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/revisiting-rock-paper-scissors#using-branches).
+1. When you're done with your feature, odds are that someone has made changes to the upstream repository in the meantime.  That means that your `main` branch is probably out of date.  Fetch the most updated copy using `$ git fetch upstream`.
+1. Now merge the upstream's changes into your local version of `main` using `$ git merge`.  Specifically, you'll first want to make sure you're on your `main` branch using `$ git checkout main` and then `$ git merge upstream/main` to merge in those upstream changes that we just fetched.  
+1. Note that a `$ git fetch upstream` followed by a `$ git merge upstream/some_branch` is the EXACT same thing as doing a `$ git pull upstream/some_branch`.  I prefer to split it up so I can explicitly walk through the steps.
```
Comment (01zulfi) [inline ruby_programming/git/lesson_using_git_in_the_real_world.md:37]:
```suggestion
1. Note that a `$ git fetch upstream` followed by a `$ git merge upstream/some_branch` is the EXACT same thing as doing a `$ git pull upstream/some_branch`.  We prefer to split it up so we can explicitly walk through the steps.
```

**Review by 01zulfi — CHANGES_REQUESTED** [review]
The changes look great!

I've left a couple of comments below. Other than that, I have a question. Is there a reason we're using the `$` sign for every git command? I know it's used to identify that the command should be run in the terminal, but I believe this is something users reading this lesson should already be familiar with. Let me know if it's a non-issue and/or any other thoughts you have regarding it.

File: ruby_programming/git/lesson_using_git_in_the_real_world.md (hunk 1 above)
Comment (wise-king-sullyman) [reply ruby_programming/git/lesson_using_git_in_the_real_world.md:17]: Great catch!

**Comment by wise-king-sullyman** [comment]
@01zulfi no reason I kept the `$` other than it was there and I didn't think to delete it. You're right, doesn't need to be there.

**Review by 01zulfi — APPROVED** [review]

**Pull request merged**